subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/testng/TestNGTestFramework.java

```diff
@@ -50,24 +47,21 @@
     private final TestNGOptions options;
     private final TestNGDetector detector;
     private final DefaultTestFilter filter;
-    private final ObjectFactory objects;
-    private final String testTaskPath;
-    private final FileCollection testTaskClasspath;
     private final Factory<File> testTaskTemporaryDir;
-    private transient ClassLoader testClassLoader;
 
     @UsedByScanPlugin("test-retry")
     public TestNGTestFramework(final Test testTask, FileCollection classpath, DefaultTestFilter filter, ObjectFactory objects) {
         this.filter = filter;
```

Comment:
It might be clearer to have the old constructor call the new one vs the new one filling in dummy values for the old one.

Reply:
OK

Reply:
@big-guy and I agreed to let this stay as-is for 7.3